Training24/testintlist.c

```diff
@@ -0,0 +1,106 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// testintlist.c
+// Program on Testing and implementing the Linked list functions.
+// -----------------------------------------------------------------------------------------------
+#include <stdio.h>
+#include "intlist.h"
+
+void PrintList (IntList* list) {
+   Node* current = list->head;
+   if (current == NULL) printf ("List is empty.\n");
+   else {
+      while (current != NULL) {
+         printf ("%d", current->data);
+         current = current->next;
+         if (current != NULL) printf (" -> ");
+      }
+      printf (" -> NULL\n");
+   }
+}
+
+void PrintError (int errorCode) {
+   switch (errorCode) {
+   case SUCCESS:
+      printf ("Operation succeeded.\n");
+      break;
+   case INVALID_INDEX:
+      printf ("Error: Invalid index.\n");
+      break;
+   case ELEMENT_NOT_FOUND:
+      printf ("Error: Element not found.\n");
+      break;
+   case MEMORY_ALLOCATION_FAILED:
+      printf ("Error: Memory allocation failed.\n");
+      break;
+   case NULL_POINTER:
+      printf ("Error: Null pointer.\n");
+      break;
+   default:
+      printf ("Error: Unknown error code %d.\n", errorCode);
+      break;
+   }
+}
+
+int main () {
+   IntList* list = Create ();
+   if (list == NULL) {
+      PrintError (GetIntListLastError ());
+      return 1;
+   }
+   printf ("List created successfully. \n");
+   printf ("\n");
+
+   Add (list, -10);
+   Add (list, 20);
+   Add (list, 30);
+   Add (list, 20);
+   Add (list, -10);
+   PrintError (GetIntListLastError ());
+   printf ("List after adding elements:\n");
+   PrintList (list);
+   printf ("\n");

```

Comment:
Add a new line while exiting the method. Don't have a separate line for this. Change this wherever necessary.